[SNMP]: Allow system with no ports in config db run without errors

liorghub opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**
Allow system with no ports in config db run without errors.
This is needed for modular system which should boot properly without line cards.

**- How I did it**
Remove snmpagent error exit if there are no ports in config DB or in counters DB.

**- How to verify it**
I perfomed snmpwalk on the root oid.

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

liorghub committed:
Add support for system without ports

dprital reviewed (commented):
Please rephrase the header of the PR to: "[SNMP]: Allow system with no ports on config_db, run without errors"

qiluo-msft committed:
Extend rfc3433.py to support more Physical Entity Sensor MIB entries

**- What I did**

Extend RFC3433 implementation with:

1. FAN tachometers
2. PSU current sensor
3. PSU voltage sensor
4. PSU power sensor
5. PSU temp sensor
6. Chassis temp sensor

MIB HLD update PR to reflect this change please refer to: https://github.com/Azure/SONiC/pull/766

A fix for the LGTM checker

**- How I did it**
1. Refactor sensor data parsing class by adding a base class  BaseSensorData;  inherit  TransceiverSensorData, PSUSensorData, FANSensorData, and ThermalSensorData from it to reduce redundant code.
2. Adding more sensor MIB entry class: PSUTempSensor, PSUVoltageSensor, PSUCurrentSensor, PSUPowerSensor, FANSpeedSensor, and ThermalSensor.
3. Separate MIB update to different functions according to different sensors types: update_xcvr_dom_data, update_psu_sensor_data, update_fan_sensor_data, and update_thermal_sensor_data.
4. Add unit test cases to cover the new added MIB entries.
5. Add lgtm.yaml to fix the LGTM checker.

**- How to verify it**

Manual test and run updated community SNMP test case(https://github.com/Azure/sonic-mgmt/pull/3357).
